source/d3d8to9_device.cpp

```diff
@@ -1060,6 +978,7 @@ HRESULT STDMETHODCALLTYPE Direct3DDevice8::GetPaletteEntries(UINT PaletteNumber,
 }
 HRESULT STDMETHODCALLTYPE Direct3DDevice8::SetCurrentTexturePalette(UINT PaletteNumber)
 {
+	PaletteNumber = (PaletteNumber & 0xFFF) | (PC_NOCOLLAPSE << 12);
```

Comment:
Yes, but those flags are for the `PALETTEENTY` structure, so I'm confused as to why adding it to the palette index changes things.